src/lib/coda_base/user_command_status.ml

```diff
@@ -24,12 +27,14 @@ module Failure = struct
         | Snapp_account_not_present
         | Update_not_permitted
         | Incorrect_nonce
-      [@@deriving sexp, yojson, eq, compare]
+      [@@deriving sexp, yojson, eq, compare, enum]
 
       let to_latest = Fn.id
     end
   end]
 
+  type failure = t
+
```

Comment:
might be clearer to add:
```ocaml
let failure_min = min
let failure_max = max
```
and use those below; the generated names `min` and `max` might appear ambiguous when used below